### PR TITLE
examples: add reflection example

### PR DESCRIPTION
**Pull request by YifeiZhuang**
Moved to a separate example folder, added a new dependency `grpc-services`

#fixit
b/259310750

### CODE REVIEW

File: examples/example-reflection/build.gradle [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        grpc { artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}" }
+    }
+    generateProtoTasks {
+        all()*.plugins { grpc {} }
+    }
+}
+
+startScripts.enabled = false
+
+task ReflectionServer(type: CreateStartScripts) {
```
Comment (YifeiZhuang) [inline examples/example-reflection/build.gradle:44]: TODO: add loop to execute all examples in buildscripts/unix.sh

Comment (YifeiZhuang) [reply examples/example-reflection/build.gradle:44]: https://github.com/grpc/grpc-java/pull/9961

File: examples/example-reflection/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+gRPC Reflection Example
+================
+
+The reflection example has a Hello World server with `ProtoReflectionService` registerred.
```
Comment (temawi) [inline examples/example-reflection/README.md:4]: "registered"

File: examples/example-reflection/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
+gRPC Reflection Example
+================
+
+The reflection example has a Hello World server with `ProtoReflectionService` registerred. 
+
+### Build the example
+
+Build the reflection registerred hello-world example server. From the `grpc-java/examples/examples-reflection`
```
Comment (temawi) [inline examples/example-reflection/README.md:8]: "registered"

File: examples/example-reflection/build.gradle [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    maven { // The google mirror is less flaky than mavenCentral()
+        url "https://maven-central.storage-download.googleapis.com/maven2/" }
+    mavenCentral()
+    mavenLocal()
+}
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+def grpcVersion = '1.53.0' // CURRENT_GRPC_VERSION
```
Comment (temawi) [inline examples/example-reflection/build.gradle:20]: Should this be the current SNAPSHOT?

**Review by temawi — APPROVED** [review]

**Pull request merged**